### PR TITLE
Add name to user

### PR DESCRIPTION
**Pull request by dsznajder**

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -34,7 +39,7 @@
       <div class="actions">
         <%= f.submit "Update" %>
       </div><br />
-      <p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete %></p>
     <% end %>
+    <p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete %></p>
```
Comment (socjopata) [inline app/views/devise/registrations/edit.html.erb]: It's good practice to format a code in a way, that I don't see a horizontal slider ;)

Comment (stevo) [inline app/views/devise/registrations/edit.html.erb]: Yup, break lines in such case..

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Admin::UsersController < ApplicationController
   def index
-    @users = User.all
+    @users = User.all.sort
```
Comment (stevo) [inline app/controllers/admin/users_controller.rb]: `sort` works fine on arrays, but on query objects prefer to use `order`

File: app/views/devise/registrations/edit.html.erb [hunk 3]
```diff
@@ -8,6 +8,11 @@
         <%= f.email_field :email, autofocus: true %>
       </div>
 
+      <div class="field">
+        <%= f.label :name %><i>(leave it if you don't want to change it)</i><br />
```
Comment (stevo) [inline app/views/devise/registrations/edit.html.erb]: Is that hint really necessary?

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,26 +6,51 @@
...
-              <li><%= link_to('Logout', destroy_user_session_path, method: :delete) %></li>
-              <li><%= link_to 'Settings', edit_user_registration_path %></li>
-              <li><%= link_to 'Comparisons', comparisons_path %></li>
-              <% if current_user.admin? %>
-                <li><%= link_to 'Users', admin_users_path %></li>
-                <li><%= link_to 'Routes', admin_routes_path %></li>
-              <% end %>
+            <li><%= link_to 'Settings', edit_user_registration_path %></li>
+
+            <% if current_user.admin? %>
```
Comment (stevo) [inline app/views/layouts/application.html.erb]: I can see chunks of duplicated code here... this is a no-go

File: spec/features/admin/route_management_spec.rb [hunk 5]
```diff
@@ -6,13 +6,14 @@
     create(:user, :admin, email: 'admin@example.com', password: 'password')
     visit '/'
 
-    click_link 'Log in'
+    find("ul[class='right hide-on-med-and-down']").click_link 'Log in'
```
Comment (stevo) [inline spec/features/admin/route_management_spec.rb]: Does not seem to be right - there should be only one clickable "Log in" link

Comment (stevo) [inline spec/features/admin/route_management_spec.rb]: And there seems to be a lot of changes like that

File: spec/features/comparison_spec.rb [hunk 6]
```diff
@@ -3,8 +3,8 @@
 describe 'Comparision', type: :feature do
   it do
     me = create(:user, email: 'sonny@example.test', password: 'password')
-    better_climber = create(:user, email: 'better_on_red_route@example.test')
-    best_climber = create(:user, email: 'better_on_red_and_green_route@example.test')
+    better_climber = create(:user, email: 'better_on_red_route@example.test', name: 'better on red')
```
Comment (stevo) [inline spec/features/comparison_spec.rb]: it seems we no longer care about email here

File: spec/models/competitors_collection_spec.rb [hunk 7]
```diff
@@ -12,15 +12,15 @@
...
   end
 
-  describe '#emails' do
-    it 'returns formated and sorted competitor emails' do
-      john = create(:user, email: 'john@example.com')
-      don = create(:user, email: 'don@example.com')
-      tony = create(:user, email: 'tony@example.com')
+  describe '#names' do
+    it 'returns formated and sorted competitor names' do
+      john = create(:user, email: 'john@example.com', name: 'John')
```
Comment (stevo) [inline spec/models/competitors_collection_spec.rb]: it seems we no longer care about emails here

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (dsznajder) [inline app/views/layouts/application.html.erb]: I wrote it like in example : http://materializecss.com/side-nav.html
That's why i scoped it in tests.

**Pull request merged**